.circleci/config.yml

```diff
@@ -2,22 +2,135 @@ version: 2.1
 
 orbs:
   artsy-remote-docker: artsy/remote-docker@volatile
+  cypress: cypress-io/cypress@3.3.1
   hokusai: artsy/hokusai@volatile
   horizon: artsy/release@volatile
   slack: circleci/slack@4.14.0
-  yarn: artsy/yarn@6.5.0
+  node: circleci/node@5.2.0
+
+executors:
+  node-executor:
+    docker:
+      - image: cimg/node:18.15.0
+
+not_main_or_staging_or_release: &not_main_or_staging_or_release
+  filters:
+    branches:
+      ignore:
+        - main
+        - staging
+        - release
+
+not_staging_or_release: &not_staging_or_release
+  filters:
+    branches:
+      ignore:
+        - staging
+        - release
+
+only_main: &only_main
+  context:
+    - hokusai
+    - slack-orb
+  filters:
+    branches:
+      only: main
+
+only_release: &only_release
+  context:
+    - hokusai
+    - slack-orb
+  filters:
+    branches:
+      only: release
 
 jobs:
-  validate_production_schema:
+  detect-secrets:
     docker:
-      - image: cimg/node:18.20.4
+      - image: artsy/detect-secrets:ci # pragma: allowlist secret
+    resource_class: small
+    working_directory: /usr/src/app
+    steps:
+      - checkout
+      - run: git ls-files -z | xargs -0 detect-secrets-hook --baseline .secrets.baseline --verbose
+
+  test:
+    executor: node-executor
+    working_directory: ~/app
+    parallelism: 6
+    steps:
+      - checkout
+      - node/install-packages:
+          pkg-manager: yarn
+      - run:
+          name: Run Jest Tests
+          command: |
+            yarn test --reporters=default --reporters=jest-junit --shard=$(expr $CIRCLE_NODE_INDEX + 1)/$CIRCLE_NODE_TOTAL
+          environment:
+            JEST_JUNIT_OUTPUT_DIR: reports
+            JEST_JUNIT_ADD_FILE_ATTRIBUTE: "true"
+      - store_artifacts:
+          path: /app/reports/junit/
+      - store_test_results:
+          path: /app/reports/junit/
+
+  smoke-tests:
+    executor: cypress/default
+    working_directory: ~/app
+    parallelism: 6
```

Comment:
Unfortunately we couldn't use the slick new cypress orb API because their parallel features require a subscription. So had to go the manual way in order to lean on CircleCI's parallelism. But things are still vastly simpler setup-wise than before.